### ISSUE TITLE
une vm pour demo

### ISSUE THREAD

**Issue by vinber** [osm-fr/infrastructure#529]
https://demo.openstreetmap.fr/map est actuellement hébergé sur un serveur perso.

le souhait est de le transférer sur un serveur de l'asso, il y a des gens partant pour administrer le serveur.

la demande est ceci :
-  RAM : 32 GB
- disque dur : 800 GB de SSD.
- CPU : CORE I7
- Il faut que. docker puisse tourner

à discuter ici @jocelynj @cquest

**Comment by vinber** [comment]
je fais une petite relance, il y a une personne qui est ok pour l'administrer.

est-ce que c'est une trop grosse demande ?

**Comment by TANK2003** [comment]
ping : nous serons deux pur administrer le serveur

**Comment by jocelynj** [comment]
L'idée est d'installer sur le nouveau cluster Moji, qui est encore en cours d'installation.

800GB de SSD, ça me parait énorme: est-ce qu'il s'agit d'une base osm2pgsql monde derrière ?

**Comment by TANK2003** [comment]
Non, il s'agit d'une base France uniquement, mais pour un rendu rapide chaque couche est pré-calculée dans une table .
Mais je viens de recalculer le poids de la base aujourd'hui = 180 GB + fichiers statiques = 20GB. 

Les 800 etait la taille de mon serveur, (apologise).

Je pense que 250 GB serait bon

**Comment by TANK2003** [comment]
@jocelynj vous avez pu avancer sur Moji ?

**Comment by Cyrille37** [comment]
Hello
On dirait que `demo.openstreetmap.fr` est hébergé sur `osm14.openstreetmap.fr`.
Mais je ne trouve pas le service sur la [page des serveurs osm_fr](https://wiki.openstreetmap.org/wiki/FR:Serveurs_OpenStreetMap_France).

**Comment by TANK2003** [comment]
Hello @Cyrille37 il est hébergé sur mon serveur perso et transit par proxy sur les serveurs OSM-FR. Le but de ce ticket est de le ramener sur un serveur OSM-FR. 
@jocelynj tu as eu du temps pour avancer sur Moji stp ?

**Comment by jocelynj** [comment]
Le cluster moji est enfin installé correctement, et j'ai commencé par y mettre le frontend osmose.

Je regarderais pour créer la VM sous peu - qui sont les admins à ajouter ?

**Comment by TANK2003** [comment]
@jocelynj super nouvelle, merci pour ton implication !
Alors ca sera deux personnes : 
- moi 
- @Cyrille37 

Je peux t'envoyer nos clés SSH s'il le faut, j'ai les deux

**Comment by Cyrille37** [comment]
hello @TANK2003 
- Te faut-il un accès à une base pour construire tes données, il doit y en avoir d'accessible dans l'infra OSM_Fr.
- Je ne me souviens pas pourquoi tu as ma clé publique ;-)

**Comment by TANK2003** [comment]
Hello, 

Je me suis emmêlé les pinceaux !! Alors : 
- C'est plus tôt vincent.lafeychine@proton.me qu'il faut ajouter et non @Cyrille37 
- Oui j'aurai besoin d'une BD postgresql + postgis;  Il faut juste qu'elle soit assez costaude, d'où la demande de RAM / CPU.
de plus, j'ai un schéma de style pour `osm_pgsql` qui peut être différent de celle que vous avez + le `hstore` qui est requis

Desolé pour la réponse tardive, je n'ai pas eu de notifs par mail, bizare

**Comment by jocelynj** [comment]
@TANK2003 : j’ai créé une VM sur le cluster moji, sur geoosm.vm.openstreetmap.fr, avec comptes pour `karl` et `lafeychine`. L’accès est en ipv6 seulement, et je n’ai pas configuré de reverse proxy pour le moment.

**Comment by TANK2003** [comment]
Merci @jocelynj , tu veux que je t'envoies ma clé SSH public par email ?

**Comment by jocelynj** [comment]
J’ai mis les clés que j’ai récupéré d’osm167.openstreetmap.fr

**Comment by TANK2003** [comment]
D'accord, j'ai changé d'ordi depuis. Je t'ai fait un mail avec ma nouvelle clé SSH

**Comment by jocelynj** [comment]
@TANK2003: j'ai mis ta nouvelle clé sur le serveur.

Par ailleurs, on a ce rôle ansible pour installer une partie ici: https://github.com/osm-fr/ansible-scripts/blob/master/roles/geoosm/tasks/main.yml. Est-ce que je le lance sur la nouvelle VM ?

**Comment by TANK2003** [comment]
Non pas besoin, merci.
Un mot de passe est demandé pour mon compte 
<img width="266" alt="image" src="https://github.com/osm-fr/infrastructure/assets/49986468/79e62019-b76a-4f8e-9068-2c1f29a7c558">
Je dois utiliser un username différent @jocelynj ?

**Comment by jocelynj** [comment]
Pourquoi tu te connectes à osm14 ?
C’est `geoosm.vm.openstreetmap.fr` qu’il faut que tu utilises.

**Comment by TANK2003** [comment]
Ahhh mais oui ! Ca marche, je suis connecté ! Super pour la RAM ! Pour le disque dur, il y'a des disques à monter ? 
<img width="335" alt="image" src="https://github.com/osm-fr/infrastructure/assets/49986468/67411827-9f3c-4e58-87de-c7034141e6de">

**Comment by jocelynj** [comment]
Bizarre, parce que j’avais configuré 300G de disque - je regarde où il est parti.

**Comment by jocelynj** [comment]
C’est bon, j’ai modifié la config pour passer à 300G de disque.

**Comment by TANK2003** [comment]
c'est parfait ! merci encore, je commence les instals et fait un retour après

**Comment by Cyrille37** [comment]
Bonjour @TANK2003 

Ça à l'air de bien fonctionner demo.openstreetmap.fr (_osm14_), ça répond rapido :-)

- Souhaites tu fermer ce ticket ?
- Et peut-être une p'tite annonce sur talk-fr@ et/ou sur le forum.osm.fr ;-)

**Comment by jocelynj** [comment]
@Cyrille37: le DNS n'a pas changé, donc je pense qu'on est encore sur l'instance externe.

@TANK2003 : tu as pu t'en sortir pour l'installation sur la nouvelle VM ?

**Comment by Cyrille37** [comment]
Salut @jocelynj 

```
$ host demo.openstreetmap.fr
demo.openstreetmap.fr is an alias for cluster-free.openstreetmap.fr.
cluster-free.openstreetmap.fr has address 213.36.253.212
cluster-free.openstreetmap.fr has IPv6 address 2a01:e0d:1:c:58bf:fac1:0:14
```

**Comment by jocelynj** [comment]
Actuellement, ça va vers un proxy, qui redirige vers portail.dataosm.info, pas chez OSM-Fr.

**Comment by cquest** [comment]
On peut fermer ?

**Comment by jocelynj** [comment]
Yep, on peut fermer, et ça n'a jamais marché correctement sur notre infra.

**Comment by TANK2003** [comment]
@jocelynj https://demo.openstreetmap.fr/ pointe encore vers une 502. Effectivement, je l'utilise uniquement en proxy vers " portail.dataosm.info"

**Comment by jocelynj** [comment]
@cquest : on dirait que demo est encore utilisé, à voir si on peut faire autrement que mettre un proxy.

**Comment by jocelynj** [comment]
@TANK2003 : le mieux serait de mettre sur un des proxys qu’on gére directement. Tu peux dire comment rediriger, est-ce que tout demo.openstreetmap.fr devrait aller vers portail.dataosm.info, ou y a-t-il des dossiers ou sous-domaines ?

**Comment by TANK2003** [comment]
Re : @jocelynj oui tout "demo.openstreetmap.fr" devrait aller vers portail.dataosm.info

Merci d'avance

**Comment by jocelynj** [comment]
J'ai mis une redirection, est-ce que ça te convient ?

**Comment by TANK2003** [comment]
@jocelynj Avant j'avais plus tôt un  reverse proxy et non une redirection comme maintenant : l'url dans le navigateur etait toujours "demo.openstreetmap.fr" 

C'est possible de l'avoir ? 

J'ai les CORS définit pour "demo.openstreetmap.fr"

**Comment by jocelynj** [comment]
Ça devrait marcher maintenant. Après, je ne sais pas trop pourquoi https://demo.openstreetmap.fr marche, mais pas http://demo.openstreetmap.fr.

**Comment by cayenne17** [comment]
> Ça devrait marcher maintenant. Après, je ne sais pas trop pourquoi https://demo.openstreetmap.fr marche, mais pas http://demo.openstreetmap.fr.

Parce que côté DEV, CORS bloque /api/group/basemaps en HTTP.
le mieux serait de faire une redirection permanente HTTP 301 HTTP 80 vers HTTPS 443.

**Comment by jocelynj** [comment]
Bonne idée, @cayenne17. J’ai du coup mis en place la redirection http->https.

@TANK2003: tu peux confirmer que tout marche correctement, et fermer le ticket ?

**Comment by TANK2003** [comment]
@jocelynj je confirme que c'est tout bon, merci !
